### PR TITLE
Remove license header from configgen

### PR DESCRIPTION
**Pull request by andreamlin**
Add comment emphasizing the optionality and default value for the `license_header` config.

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/configgen/mergers/ConfigMerger.java [hunk 1]
```diff
@@ -75,7 +73,7 @@ private FieldConfigNode mergeConfig(ApiModel model, FieldConfigNode configNode)
     ConfigNode languageSettingsNode =
         languageSettingsMerger.mergeLanguageSettings(packageName, configNode, versionNode);
 
-    mergeLicenseHeader(configNode, languageSettingsNode);
+    // mergeLicenseHeader(configNode, languageSettingsNode);
```
Comment (michaelbausor) [inline src/main/java/com/google/api/codegen/configgen/mergers/ConfigMerger.java]: Just delete this line?

File: src/main/proto/com/google/api/codegen/config.proto [hunk 2]
```diff
@@ -144,6 +144,7 @@ message LicenseHeaderProto {
   string copyright_file = 1; // DEPRECATED. The Google copyright will always be used.
 
   // The file containing the raw license header without any copyright line(s).
+  // OPTIONAL. If not value is not given, the Apache 2.0 license will be used.
```
Comment (michaelbausor) [inline src/main/proto/com/google/api/codegen/config.proto]: s/not value is/value is not

File: src/main/java/com/google/api/codegen/configgen/mergers/ConfigMerger.java (hunk 1 above)
Comment (andreamlin) [reply src/main/java/com/google/api/codegen/configgen/mergers/ConfigMerger.java]: done

File: src/main/proto/com/google/api/codegen/config.proto (hunk 2 above)
Comment (andreamlin) [reply src/main/proto/com/google/api/codegen/config.proto]: done

**Pull request merged**